kayenta-judge/src/main/scala/com/netflix/kayenta/judge/NetflixACAJudge.scala

```diff
@@ -63,14 +63,26 @@ class NetflixACAJudge extends CanaryJudge with StrictLogging {
   /**
     * Metric Transformations
     */
-  def transformMetric(metric: Metric, nanStrategy: NaNStrategy): Metric = {
-    val detector = new IQRDetector(factor = 3.0, reduceSensitivity = true)
-    val transform = if (nanStrategy == NaNStrategy.Remove) {
-      Function.chain[Metric](Seq(Transforms.removeNaNs(_), Transforms.removeOutliers(_, detector)))
+  def transformMetric(metric: Metric, nanStrategy: NaNStrategy, outlierStrategy: OutlierStrategy): Metric = {
+    handleOutliers(handleNaNs(metric, nanStrategy), outlierStrategy)
+  }
+
+  def handleNaNs(metric: Metric, nanStrategy: NaNStrategy): Metric = {
+    if (nanStrategy == NaNStrategy.Remove) {
+      Transforms.removeNaNs(metric)
+    } else {
+      Transforms.replaceNaNs(metric)
+    }
+  }
+
+  def handleOutliers(metric: Metric, outlierStrategy: OutlierStrategy): Metric = {
+    if (outlierStrategy == OutlierStrategy.Remove) {
+      val detector = new IQRDetector(factor = 3.0, reduceSensitivity = true)
```

Comment:
[Consideration] Given that you are adding configuration, would you also want to consider making the IQR factor configurable?

Reply:
Yep, done.

This is how the configuration now looks with both optional values specified:

```
      "analysisConfigurations": {
        "canary": {
          "outliers": {
            "strategy": "remove",
            "iqrFactor": 2.75
          }
        }
      },
```

Reply:
We really need to document this somewhere.

I have been using this gist when peopler ask: https://gist.github.com/fieldju/ebfc89b23f6ce66679100d30bf536d91

Reply:
Maybe we can move that markdown gist into the project and add this new feature to it, so that we can link people to it.

Reply:
That's a great write-up. Can you please open a PR to add that into the project?

Reply:
I will work on it, and tweak the language to be more for all users rather than just my org.

Reply:
I opened that PR, when that is merged, I would appreciate it if you added your new stuff ...

```json
          "outliers": {
            "strategy": "remove",
            "iqrFactor": 2.75
          }
```

to the analysis configurations section.

Reply:
Yes, will do.

Reply:
I would like to propose renaming the `iqrFactor` property to `outlierFactor`.

Reply:
I have no objection to renaming the property. @gregorymfoster What say you?

Reply:
I've renamed the property.